src/Umbraco.Web/PublishedCache/NuCache/DataSource/DatabaseDataSource.cs

```diff
@@ -64,8 +69,50 @@ private Sql<ISqlContext> ContentSourcesSelect(IScope scope, Func<Sql<ISqlContext
 
             return sql;
         }
+        private static Sql<ISqlContext> PublishedOnlyContentSourcesSelect(IScope scope, Func<Sql<ISqlContext>, Sql<ISqlContext>> joins = null)
+        {
+            var sql = scope.SqlContext.Sql()
+                .Select<NodeDto>(x => Alias(x.NodeId, "Id"), x => Alias(x.UniqueId, "Uid"),
+    x => Alias(x.Level, "Level"), x => Alias(x.Path, "Path"), x => Alias(x.SortOrder, "SortOrder"), x => Alias(x.ParentId, "ParentId"),
+    x => Alias(x.CreateDate, "CreateDate"), x => Alias(x.UserId, "CreatorId"))
+                .AndSelect<ContentDto>(x => Alias(x.ContentTypeId, "ContentTypeId"))
+                .AndSelect<DocumentDto>(x => Alias(x.Published, "Published"), x => Alias(x.Edited, "Edited"))
+
+            .AndSelect<ContentVersionDto>(x => Alias(x.Id, "VersionId"), x => Alias(x.Text, "EditName"), x => Alias(x.VersionDate, "EditVersionDate"), x => Alias(x.UserId, "EditWriterId"))
+            .AndSelect<DocumentVersionDto>(x => Alias(x.TemplateId, "EditTemplateId"))
+
+            .AndSelect<ContentVersionDto>("pcver", x => Alias(x.Id, "PublishedVersionId"), x => Alias(x.Text, "PubName"), x => Alias(x.VersionDate, "PubVersionDate"), x => Alias(x.UserId, "PubWriterId"))
+            .AndSelect<DocumentVersionDto>("pdver", x => Alias(x.TemplateId, "PubTemplateId"))
+
+            .AndSelect<ContentNuDto>("nuPub", x => Alias(x.Data, "PubData"))
+
+
+            .From<NodeDto>();
+
+
+            // TODO: I'm unsure how we can format the below into SQL templates also because right.Current and right.Published end up being parameters
```

Comment:
My memory is blurred on this one but I feel like we solved this with another one of your PRs a while ago? Not sure if you remember or not. We can always deal with that later though.